python/tchannel/messages/call_request.py

```diff
@@ -34,17 +37,24 @@ class CallRequestMessage(BaseMessage):
         'arg_3',
     )
 
+    CHECKSUM = {
+        0x00: 0,
+        0x01: 4,
+        0x02: 4
+    }
+
     FLAGS_SIZE = 1
     TTL_SIZE = 4
+    # sizeOf(span_id) + sizeOf(parent_id) + sizeOf(trace_id) = 3*8 = 24
     TRACE_SIZE = 8
     TRACEFLAGS_SIZE = 1
     SERVICE_SIZE = 2
     NH_SIZE = 1
     HEADER_SIZE = 1
     CSUMTYPE_SIZE = 1
-    CSUM_SIZE = 4
+    CSUM_SIZE = CHECKSUM[CSUMTYPE_SIZE]
```

Comment:
I don't think you can do this at import time -- it's going to vary based on the message.